ui/src/buckets/components/UpdateBucketOverlay.tsx

```diff
@@ -1,153 +1,159 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useState,
+  ChangeEvent,
+  FormEvent,
+} from 'react'
 import {withRouter, WithRouterProps} from 'react-router'
 import {connect} from 'react-redux'
+import {get} from 'lodash'
 
 // Components
-import {Overlay} from '@influxdata/clockface'
+import {
+  Overlay,
+  RemoteDataState,
+  SpinnerContainer,
+  TechnoSpinner,
+} from '@influxdata/clockface'
 import BucketOverlayForm from 'src/buckets/components/BucketOverlayForm'
 
 // Actions
 import {updateBucket} from 'src/buckets/actions'
+import {notify} from 'src/shared/actions/notifications'
+
+// APIs
+import * as api from 'src/client'
 
 // Constants
 import {DEFAULT_SECONDS} from 'src/buckets/components/Retention'
+import {getBucketFailed} from 'src/shared/copy/notifications'
 
-// Types
-import {AppState, Bucket} from 'src/types'
-
-interface State {
-  bucket: Bucket
-  ruleType: 'expire'
-}
-
-interface StateProps {
-  bucket: Bucket
-}
+//Types
+import {Bucket} from 'src/types'
 
 interface DispatchProps {
   onUpdateBucket: typeof updateBucket
+  onNotify: typeof notify
 }
 
-type Props = StateProps & DispatchProps & WithRouterProps
-
-class UpdateBucketOverlay extends PureComponent<Props, State> {
-  constructor(props) {
-    super(props)
-
-    const {bucket} = this.props
-
-    this.state = {
-      ruleType: this.ruleType(bucket),
-      bucket,
-    }
-  }
+type Props = DispatchProps & WithRouterProps
 
-  public render() {
-    const {bucket, ruleType} = this.state
+const UpdateBucketOverlay: FunctionComponent<Props> = ({
+  onUpdateBucket,
+  onNotify,
+  params: {bucketID, orgID},
+  router,
+}) => {
+  const [bucketDraft, setBucketDraft] = useState<Bucket>(null)
 
-    return (
-      <Overlay visible={true}>
-        <Overlay.Container maxWidth={500}>
-          <Overlay.Header title="Edit Bucket" onDismiss={this.handleClose} />
-          <Overlay.Body>
-            <BucketOverlayForm
-              name={bucket.name}
-              buttonText="Save Changes"
-              ruleType={ruleType}
-              onCloseModal={this.handleClose}
-              onSubmit={this.handleSubmit}
-              disableRenaming={true}
-              onChangeInput={this.handleChangeInput}
-              retentionSeconds={this.retentionSeconds}
-              onChangeRuleType={this.handleChangeRuleType}
-              onChangeRetentionRule={this.handleChangeRetentionRule}
-            />
-          </Overlay.Body>
-        </Overlay.Container>
-      </Overlay>
-    )
-  }
+  const [loadingStatus, setLoadingStatus] = useState(RemoteDataState.Loading)
 
-  private get retentionSeconds(): number {
-    const rule = this.state.bucket.retentionRules.find(r => r.type === 'expire')
+  const [retentionSelection, setRetentionSelection] = useState(DEFAULT_SECONDS)
 
-    if (!rule) {
-      return DEFAULT_SECONDS
-    }
+  useEffect(() => {
+    const fetchBucket = async () => {
+      const resp = await api.getBucket({bucketID})
```

Comment:
it seems like the friction to this pattern would be that this function isn't cached. for overlays, there is only going to be one per page at a time, but when building out a reusable pattern of `component owns its own state`, certain affordances need to be made around cache invalidation/expiration. not going to use this PR as a springboard for that discussion, just something to start thinking about (multiple components using the same api transform within their useEffect lifecycles)